src/amo/components/Search/index.js

```diff
@@ -60,49 +85,73 @@ export class SearchBase extends React.Component {
       LinkComponent,
       count,
       enableSearchSort,
+      errorHandler,
       filters,
-      hasSearchParams,
       loading,
-      page,
+      paginationQueryParams,
       pathname,
       results,
     } = this.props;
-    const queryParams = this.props.queryParams ||
+
+    const page = parsePage(filters.page);
+
+    // We allow specific paginationQueryParams instead of always using
+    // convertFiltersToQueryParams(filters) so certain search filters
+    // aren't repeated if they are elsewhere in the URL. This is useful
+    // for pages like the category page which contain `addonType` and
+    // `category` in their URLs
+    // (eg: `/extensions/categories/feed-news-blogging/`) so they don't
+    // need them in the queryParams.
+    //
+    // If paginator params aren't specified, we fallback to filters.
```

Comment:
Oh, yeah, that is a bit cleaner. The only one to worry about is the implicit `page=1` but paginator will always sort that out. As you say we'll leave this for now but I've filed: #2937